tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py

```diff
@@ -4829,4 +4829,26 @@ def test_allowed_import_else(self):
             self.checker.visit_importfrom(ima)
             self.checker.visit_import(imb)
             self.checker.visit_import(imc)
-            self.checker.visit_importfrom(imd)
\ No newline at end of file
+            self.checker.visit_importfrom(imd)
+
+class TestCheckDoNotUseLegacyTyping(pylint.testutils.CheckerTestCase):
+    """Test that we are blocking disallowed legacy typing practices"""
+
+    CHECKER_CLASS = checker.DoNotUseLegacyTyping
+
+    def test_disallowed_typing(self):
+        """Check that illegal method typing comments raise warnings"""
+        fnt = astroid.extract_node(
+            """
+            def function(x):
+                # type: (str) -> str #@
+                pass
+            """
+        )
+        with self.assertAddsMessages(
```

Comment:
Is it possible to add a negative test for `self.assertNoMessages` for a function that has been typed correctly?

Reply:
As it's written, no, as the ast [node](https://docs.python.org/3/library/ast.html#ast.FunctionType) used for the check (`FunctionType`) is exclusive to the legacy type hinting comments. Unit or functional tests could work, but fall outside of the pytest framework currently in use.